src/prototype_room_creepbuilder.js

```diff
@@ -94,7 +94,18 @@ Room.prototype.spawnCheckForCreate = function(creepsConfig) {
         }
         return 4;
       }
-
+      // add external supply in front of the other
+            if (target !== room.name) {
+              if (object.role == 'carry' ) {
+                return 4;
```

Comment:
I think I would start at 5. Having in room creeps more important than external.

E.g. if a room is under attack, it is more important that these creeps are spawned, than an external carry. Maybe discussable